File: scripts/sync-prices-from-csv.py
#!/usr/bin/env python3
import csv
import json
import firebase_admin
from firebase_admin import credentials, firestore
from datetime import datetime

# Initialize Firebase
cred = credentials.Certificate('bebias-chatbot-key.json')
firebase_admin.initialize_app(cred)
db = firestore.client()

print('📥 Reading WooCommerce CSV export...\n')

csv_path = '/Users/giorginozadze/Downloads/wc-product-export-20-11-2025-1763612693038.csv'

products_to_update = []

with open(csv_path, 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f)

    for row in reader:
        product_type = row.get('Type', '').lower()
        name = row.get('Name', '').strip()
        price_str = row.get('Regular price', '').strip()
        stock_str = row.get('Stock', '').strip()
        product_id = row.get('ID', '').strip()

        if not name:
            continue

        # Only variations and simple products
        if product_type not in ['variation', 'simple']:
            continue

        # Parse price
        try:
            price = float(price_str) if price_str else 0
        except ValueError:
            price = 0

        # Parse stock
        try:
            stock = int(stock_str) if stock_str else 0
        except ValueError:
            stock = 0

        if price > 0:
            products_to_update.append({
                'id': product_id,
                'name': name,
                'type': product_type,
                'price': price,
                'stock': stock
            })

print(f'Found {len(products_to_update)} products with prices in CSV\n')

# Get all Firestore products
firestore_products = {}
products_ref = db.collection('products')
docs = products_ref.stream()

for doc in docs:
    data = doc.to_dict()
    firestore_products[doc.id] = {'id': doc.id, **data}

    # Also index by name for easier matching
    if 'name' in data:
        firestore_products[data['name']] = {'id': doc.id, **data}

print(f'Firestore products: {len(set([p["id"] for p in firestore_products.values()]))}\n')
print('🔄 Updating prices...\n')

updated = 0
skipped = 0
not_found = 0

for csv_product in products_to_update:
    name = csv_product['name']
    price = csv_product['price']
    csv_id = csv_product['id']

    # Try to find in Firestore by name or ID
    firestore_product = firestore_products.get(name) or firestore_products.get(csv_id)

    if not firestore_product:
        print(f'⚠️  Not found: {name}')
        not_found += 1
        continue

    # Check if update needed
    current_price = firestore_product.get('price', 0)

    if current_price == price:
        skipped += 1
        continue

    # Update price in Firestore
    try:
        doc_ref = products_ref.document(firestore_product['id'])
        doc_ref.update({
            'price': price,
            'currency': 'GEL',
            'last_updated': datetime.now().isoformat(),
            'last_updated_by': 'csv_price_sync'
        })

        print(f'✅ {name}: {current_price} → {price} GEL')
        updated += 1
    except Exception as e:
        print(f'❌ Failed to update {name}: {e}')

print(f'\n📊 Summary:')
print(f'   Updated: {updated}')
print(f'   Skipped (no change): {skipped}')
print(f'   Not found: {not_found}')
print(f'\n✅ Price sync complete!')
print('\nNext step: Run sync to update products.json')
print('  node scripts/sync-from-firestore.js')

File: scripts/sync-prices-from-csv.js
#!/usr/bin/env node
const admin = require('firebase-admin');
const path = require('path');
const fs = require('fs');

// Initialize Firebase Admin with JSON key file
if (!admin.apps.length) {
  const serviceAccountPath = path.join(__dirname, '..', 'bebias-chatbot-key.json');
  const serviceAccount = require(serviceAccountPath);
  admin.initializeApp({
    credential: admin.credential.cert(serviceAccount)
  });
}

const db = admin.firestore();

function parseCSV(filePath) {
  const content = fs.readFileSync(filePath, 'utf8');
  const lines = content.split('\n');
  const headers = lines[0].split(',').map(h => h.replace(/"/g, '').trim());

  const products = [];

  for (let i = 1; i < lines.length; i++) {
    if (!lines[i].trim()) continue;

    // Simple CSV parsing (handles quoted fields)
    const values = [];
    let current = '';
    let inQuotes = false;

    for (let char of lines[i]) {
      if (char === '"') {
        inQuotes = !inQuotes;
      } else if (char === ',' && !inQuotes) {
        values.push(current.trim());
        current = '';
      } else {
        current += char;
      }
    }
    values.push(current.trim());

    const product = {};
    headers.forEach((header, index) => {
      product[header] = values[index] || '';
    });

    products.push(product);
  }

  return products;
}

async function syncPricesFromCSV() {
  const csvPath = '/Users/giorginozadze/Downloads/wc-product-export-20-11-2025-1763612693038.csv';

  console.log('📥 Reading WooCommerce CSV export...\n');

  try {
    const csvProducts = parseCSV(csvPath);
    console.log(`Found ${csvProducts.length} products in CSV\n`);

    // Filter only variations and simple products with prices
    const productsWithPrices = csvProducts.filter(p => {
      const type = p.Type?.toLowerCase();
      const price = parseFloat(p['Regular price']) || 0;
      return (type === 'variation' || type === 'simple') && price > 0;
    });

    console.log(`Products with prices (variation/simple): ${productsWithPrices.length}\n`);

    // Get all Firestore products
    const firestoreSnapshot = await db.collection('products').get();
    const firestoreProducts = new Map();

    firestoreSnapshot.forEach(doc => {
      const data = doc.data();
      firestoreProducts.set(doc.id, { id: doc.id, ...data });

      // Also index by name for easier matching
      if (data.name) {
        firestoreProducts.set(data.name, { id: doc.id, ...data });
      }
    });

    console.log(`Firestore products: ${firestoreSnapshot.size}\n`);
    console.log('🔄 Updating prices...\n');

    let updated = 0;
    let skipped = 0;
    let notFound = 0;

    for (const csvProduct of productsWithPrices) {
      const name = csvProduct.Name;
      const price = parseFloat(csvProduct['Regular price']);
      const stock = parseInt(csvProduct.Stock) || 0;
      const id = csvProduct.ID;

      if (!name || price <= 0) {
        skipped++;
        continue;
      }

      // Try to find in Firestore by name or ID
      let firestoreProduct = firestoreProducts.get(name) || firestoreProducts.get(id);

      if (!firestoreProduct) {
        console.log(`⚠️  Not found: ${name}`);
        notFound++;
        continue;
      }

      // Check if update needed
      const currentPrice = firestoreProduct.price || 0;
      if (currentPrice === price) {
        skipped++;
        continue;
      }

      // Update price in Firestore
      try {
        await db.collection('products').doc(firestoreProduct.id).update({
          price: price,
          currency: 'GEL',
          last_updated: new Date().toISOString(),
          last_updated_by: 'csv_price_sync'
        });

        console.log(`✅ ${name}: ${currentPrice} → ${price} GEL`);
        updated++;
      } catch (error) {
        console.error(`❌ Failed to update ${name}:`, error.message);
      }
    }

    console.log(`\n📊 Summary:`);
    console.log(`   Updated: ${updated}`);
    console.log(`   Skipped (no change): ${skipped}`);
    console.log(`   Not found: ${notFound}`);
    console.log(`\n✅ Price sync complete!`);
    console.log('\nNext step: Run sync to update products.json');
    console.log('  node scripts/sync-from-firestore.js');

    process.exit(0);
  } catch (error) {
    console.error('❌ Error:', error);
    process.exit(1);
  }
}

syncPricesFromCSV();
